client/my-sites/plugins/plugin-activate-toggle/test/index.jsx

```diff
@@ -1,83 +1,67 @@
  */
-var chai = require( 'chai' ),
-	expect = chai.expect,
-	ReactDom = require( 'react-dom' ),
-	React = require( 'react' ),
-	mockery = require( 'mockery' ),
-	sinon = require( 'sinon' ),
-	noop = require( 'lodash/noop' ),
-	TestUtils = require( 'react-addons-test-utils' );
+import { expect } from 'chai';
+import identity from 'lodash/identity';
+import mockery from 'mockery';
+import { mount } from 'enzyme';
+import React from 'react';
+import { spy } from 'sinon';
 
 /**
- * Mocks & fixtures
+ * Internal dependencies
  */
-var mockedPluginAction = require( './mocks/plugin-action' ),
-	fixtures = require( './fixtures' ),
-	mockedActions = require( './mocks/actions' );
-
-require( 'lib/react-test-env-setup' )();
+import EmptyComponent from 'test/helpers/react/empty-component';
+import fixtures from './fixtures';
+import mockedActions from './mocks/actions';
+import mockedPluginAction from './mocks/plugin-action';
+import useFakeDom from 'test/helpers/use-fake-dom';
+import useMockery from 'test/helpers/use-mockery';
 
 describe( 'PluginActivateToggle', function() {
-	var PluginActivateToggle, analyticsMock;
-
-	analyticsMock = {
-		ga: { recordEvent: sinon.spy() },
-		tracks: { recordEvent: sinon.spy() }
+	const analyticsMock = {
+		ga: { recordEvent: spy() },
+		tracks: { recordEvent: spy() }
 	};
+	let PluginActivateToggle;
+
+	useFakeDom();
+	useMockery();
 
 	before( function() {
 		mockery.registerMock( 'analytics', analyticsMock );
 		mockery.registerMock( 'my-sites/plugins/plugin-action/plugin-action', mockedPluginAction );
 		mockery.registerMock( 'lib/plugins/actions', mockedActions );
-		mockery.registerMock( 'component-classes', function() {
-			return { add: noop, toggle: noop, remove: noop }
-		} );
-		mockery.registerSubstitute( 'matches-selector', 'component-matches-selector' );
-		mockery.registerSubstitute( 'query', 'component-query' );
-		mockery.enable( {
-			warnOnReplace: false,
-			warnOnUnregistered: false
-		} );
-	} );
+		mockery.registerMock( 'my-sites/plugins/disconnect-jetpack/disconnect-jetpack-button', EmptyComponent );
 
-	beforeEach( function() {
-		this.timeout( 10 * 1000 );
 		PluginActivateToggle = require( 'my-sites/plugins/plugin-activate-toggle' );
-		PluginActivateToggle.prototype.translate = function( str ) {
-			return str;
-		};
+		PluginActivateToggle.prototype.translate = identity;
 	} );
 
 	afterEach( function() {
-		ReactDom.unmountComponentAtNode( document.body );
 		mockedActions.togglePluginActivation.reset();
 		analyticsMock.ga.recordEvent.reset();
 	} );
 
 	it( 'should render the component', function() {
-		var rendered = TestUtils.renderIntoDocument( <PluginActivateToggle { ...fixtures } /> ),
-			pluginActionToggle = TestUtils.scryRenderedDOMComponentsWithClass( rendered, 'plugin-action' );
+		const wrapper = mount( <PluginActivateToggle { ...fixtures } /> );
 
-		expect( pluginActionToggle.length ).to.equal( 1 );
+		expect( wrapper.find( '.plugin-action' ) ).to.have.length( 1 );
```

Comment:
This is what I suggested initially from my own recollection, and should work fine, but I found to be deprecated and will be removed in a future version.

>Deprecation notice: Using `length` as an assertion will be deprecated in version 2.4.0 and removed in 3.0.0. Code using the old style of asserting for `length` property value using `length(value)` should be switched to use `lengthOf(value)` instead.

http://chaijs.com/api/bdd/#method_length

There's probably other instances of this we'll need to fix when we upgrade anyways, so not urgent but rather an FYI.